File: raspberry-pi-ios-client/RaspberryPiClient/Data/Infrastructure/Ros/Topic/Message/Geometry/TwistMessage.swift
//
//  TwistMessage.swift
//  RaspberryPiClient
//
//  Created by Shusuke Ota on 2025/12/29.
//

import Foundation

/// [geometry_msgs/msg/Twist.msg](https://github.com/ros2/common_interfaces/blob/jazzy/geometry_msgs/msg/Twist.msg)
struct TwistMessage: RosMessageProtocol {
    static var rosSchemaName: String { "geometry_msgs/msg/Twist" }

    /// translational speed
    let linear: Vector3Message
    /// angular velocity
    let angular: Vector3Message
}

File: raspberry-pi-android-client/app/src/main/java/com/shusuke/raspberry_pi_android_client/data/infrastructure/ros/topic/message/TwistMessage.kt
package com.shusuke.raspberry_pi_android_client.data.infrastructure.ros.topic.message

import kotlinx.serialization.Serializable

/**
 * [geometry_msgs/msg/Twist.msg](https://github.com/ros2/common_interfaces/blob/jazzy/geometry_msgs/msg/Twist.msg)
 */
@Serializable
data class TwistMessage(
    /** 並進速度 */
    val linear: Vector3Message = Vector3Message(),
    /** 角速度 */
    val angular: Vector3Message = Vector3Message()
)
